tests/simulation/test_backend.py

```diff
@@ -59,17 +71,22 @@ def test_backend_x(nqubits, Backend):
 def test_backend_result_x_and_many_cnot(nqubits, Backend):
 
     # Create circuit that maps |0..0> to |1...1> with one X and many CNOTs.
-    mp_list = [[X] + [identity for i in range(nqubits - 1)]]
-    for j in range(nqubits - 1):
-        mp_list.append([identity for i in range(j)] + [CNOT, 1] + [identity for i in range(nqubits - 2 - j)])
-
     tb = Backend(nqubit=nqubits)
-
-    # Check that it does the correct mapping
     psi0 = np.zeros(2**nqubits)
     psi0[0] = 1.0
-
-    psi1 = tb.statevector(mp_list, psi0)
+    if isinstance(tb, BinaryBackend):
```

Comment:
@paolodr98 Let us have chat about the change in the interface between the BinaryBackend and the other backends. It would be great if they were 1:1 replacements of each other.